main.cpp

```diff
@@ -1,23 +1,36 @@
-#include "queue.h"
 #include <iostream>
+#include "queue.h"
 using namespace std;
-int main()
-{
-	int c, b;
-	data a[12], d;
-	ArrayQueue q(20);
-	cout << "press 1 if you want ArrayQueue, press 2 if you want ListQueue";
-	cin >> b;
-	if (b == 2)
-		ListQueue q;
-	for (c = 0;c < 10;c++)
-	{
-		cin >> d;
-		q.enqueue(d);
-	
-	}
-	for (c = 0;c < 10;c++)
-	{
-		cout << q.denqueue() << " ";
-	}
+
+ int main()
+ {
+    char c;
+    std::cout << "Type of queue (a - array, l - list): ";
+    std::cin >> c;
+	ArrayQueue queue(20);
+    if (c == 'a')
+	ArrayQueue queue(20);
+    else if (c == 'l')
+        ListQueue queue;

```

Comment:
Do you think this line has any sense?
This is just a definition of a new variable with the same name as variable at line 10. It will work only inside if () {} block.
Please, carefully inspect code from template main(). There we have `BasicQueue *queue` as a pointer of the base class.